### ISSUE TITLE
Eager validation in MTextField overrides property buffering!

### ISSUE THREAD

**Issue by mwojterski** [viritin/viritin#174]
When _eagerValidation_ is enabled in `MTextField` (which is default for `AbstractForm`), any value change that is valid is immediately propagated to property data source.

There is a comment included in `doEagerValidation` method:
```
// Also eagerly pass content to backing bean to make top level 
// validation eager, but do not listen the value back in value change
// event
```

I understand this is done to allow `MValidator`s in `AbstraftForm` operate on updated beans, but please consider that in normal flow, even if data entered into the form is valid, user may still want to _discard_ changes. This use case is where _buffering_ comes in.

Therefore I'd suggest running `MValidator`s on proxy which would intercept getters and use property data sources and throw for any setter (_I guess validator should not modify underlying bean, right?_).

Also, please note, that `IntegerField` uses normal `setValue` method in `userInputToValue` therefore buffering still works, but `MValidator`s will operate on stale values.

**Comment by mstahv** [comment]
Hi this is a pretty common thing I'm discussing with a variety of persons, but after each discussion I'm just more confident the the buffering feature shouldn't exist in the framework at all. I actually recently wrote a [blog post](https://vaadin.com/web/matti/blog/-/blogs/3-pro-tips-for-vaadin-developers) about this as well.

What kind of domain objects are you using? Most often they are "detached" so when resetting the value, you can just fetch a fresh entity from your backend. If your entities really are "live", you can most often make a clone of it (if not cloneable, then by e.g. with apache beanutils) or even go as far as creating a DTO and clone the state of your live object to it, and then back in save. Even the last option is rather easy and fast to implement with modern IDEs and helper libraries.

If you have more details about your use case, I'd be happy to listen. But I think this would just need more detailed examples and warnings to the API.

**Comment by mwojterski** [comment]
Hi, shortly after I posted this issue, I "_fixed_" buffering by using `MTextField` subclass and then I've realized that my fix to buffering, actually _broke_ JSR bean validation... After that I did exactly what you just proposed and introduced clone-like method (_simple clone but without an ugly exception_) and manually handled resetting the bean. Now that you've mentioned that solution, I feel more confident it's a right way to do this.

Great post by the way! I wish I've seen it earlier! Funny thing, it has a comment from this year, in which someone is also complaining about the buffering. Anyway, I must say I've always admired the effort invested into this whole property-item stuff and since buffering is part of it, it seemed wasteful not to use it, but if you put it that way, I can totally live without it, especially if it's hindering you from introducing improvements. Viritin (Maddon) is a great stuff and I'm very grateful! :+1:
